nsv13/code/modules/overmap/traders_items.dm

```diff
@@ -25,9 +28,123 @@
 		SEND_SOUND(user, 'nsv13/sound/effects/ship/freespace2/computer/textdraw.wav')
 		to_chat(user, "<span class='boldnotice'>[pick(inhabited_trader.greetings)]</span>")
 
-//Nope!
+/obj/structure/overmap/trader/proc/add_objective( objective )
+	if ( objective )
+		expecting_cargo += objective 
+		essential = TRUE
+
+/obj/structure/overmap/trader/receive_cargo( mob/living/user, var/obj/machinery/computer/ship/dradis/cargo/console, var/obj/item/ship_weapon/ammunition/torpedo/freight/shipment )
+	if ( inhabited_trader )
+		var/datum/freight_delivery_receipt/receipt = new /datum/freight_delivery_receipt()
+		receipt.courier = user
+		receipt.vessel = console.linked 
+		receipt.shipment = shipment
+		receipts += receipt
+
+		check_objectives( receipt )
+	else 
+		. = ..() // Fallback to attempting a physical delivery to a station zlevel, if it exists 
+	
+/obj/structure/overmap/trader/proc/check_objectives( var/datum/freight_delivery_receipt/receipt )
```

Comment:
Could we please add a delay to this to at least simulate travel time/checking what's in the torp?